### ISSUE TITLE
FNX3-16677 ⁃ [Bug] Opening 2 tabs sent from another device crashes the app

### ISSUE THREAD

**Issue by sv-ohorvath** [mozilla-mobile/fenix#2916]
## Steps to reproduce
1. Login on 2 devices with the same account.
1. Send 2 tabs individually, from one device to another.
1. Open the first tab received, then open the second.
    1. Expected behavior
Both tabs are opened.
    1. Actual behavior
App crashes
[logcat.txt](https://github.com/mozilla-mobile/fenix/files/3227529/logcat.txt)

### Device information
* Android device: Pixel (Android 9), HTC Desire 820 (Android 6)
* Fenix version: 5/28

**Comment by jonalmeida** [comment]
From the logcat, the display acquired bug:
```
    --------- beginning of crash
2019-05-28 14:59:20.379 15350-15350/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.mozilla.fenix, PID: 15350
    java.lang.RuntimeException: Unable to resume activity {org.mozilla.fenix/org.mozilla.fenix.HomeActivity}: java.lang.RuntimeException: Failed to call observer method
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3822)
        at android.app.ActivityThread.performNewIntents(ActivityThread.java:3172)
        at android.app.ActivityThread.handleNewIntent(ActivityThread.java:3179)
        at android.app.servertransaction.NewIntentItem.execute(NewIntentItem.java:49)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.RuntimeException: Failed to call observer method
        at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:6)
        at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeMethodsForEvent(ClassesInfoCache.java:2)
        at androidx.lifecycle.ReflectiveGenericLifecycleObserver.onStateChanged(ReflectiveGenericLifecycleObserver.java:2)
        at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:3)
        at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:48)
        at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:5)
        at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:2)
        at androidx.fragment.app.Fragment.performStart(Fragment.java:10)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:105)
        at androidx.fragment.app.FragmentManagerImpl.addAddedFragments(FragmentManagerImpl.java:6)
        at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManagerImpl.java:54)
        at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManagerImpl.java:11)
        at androidx.fragment.app.FragmentManagerImpl.execPendingActions(FragmentManagerImpl.java:4)
        at androidx.fragment.app.Fragment.performResume(Fragment.java:3)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:108)
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManagerImpl.java:9)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:209)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManagerImpl.java:2)
        at androidx.fragment.app.FragmentManagerImpl.dispatchResume(FragmentManagerImpl.java:3)
        at androidx.fragment.app.FragmentActivity.onResumeFragments(FragmentActivity.java:3)
        at androidx.fragment.app.FragmentActivity.onPostResume(FragmentActivity.java:2)
        at androidx.appcompat.app.AppCompatActivity.onPostResume(AppCompatActivity.java:1)
        at android.app.Activity.performResume(Activity.java:7325)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3814)
        at android.app.ActivityThread.performNewIntents(ActivityThread.java:3172) 
        at android.app.ActivityThread.handleNewIntent(ActivityThread.java:3179) 
        at android.app.servertransaction.NewIntentItem.execute(NewIntentItem.java:49) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6718) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
     Caused by: java.lang.IllegalStateException: Display already acquired
        at org.mozilla.geckoview.GeckoSession.acquireDisplay(GeckoSession.java:7)
        at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:19)
        at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:5)
        at mozilla.components.browser.engine.gecko.GeckoEngineView.render(GeckoEngineView.kt:9)
        at mozilla.components.feature.session.EngineViewPresenter.renderSession$feature_session_release(EngineViewPresenter.kt:1)
        at mozilla.components.feature.session.SessionFeature.start(SessionFeature.kt:7)
2019-05-28 14:59:20.379 15350-15350/? E/AndroidRuntime:     at mozilla.components.support.base.feature.ViewBoundFeatureWrapper.start$support_base_release(ViewBoundFeatureWrapper.kt:1)
        at mozilla.components.support.base.feature.LifecycleBinding.start(ViewBoundFeatureWrapper.kt:1)
        at java.lang.reflect.Method.invoke(Native Method)
        at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:4)
        	... 35 more
```

**Comment by softvision-miralobontiu** [comment]
Verified as fixed on the latest build 1.0.1923 (#11560607), with Xiaomi Mi 8 Lite (Android 8.1), and HTC 10 (Android 8).
